scripts/functions_deploy.py

```diff
@@ -71,12 +72,25 @@ def main(args):
     VERBOSE = parsedArgs.verbose
 
     namespace = getRequiredParameter(config, 'cloudfunctions_namespace')
-    username = getRequiredParameter(config, 'cloudfunctions_username')
-    password = getRequiredParameter(config, 'cloudfunctions_password')
+    apikey = getOptionalParameter(config, 'cloudfunctions_apikey')
+    username = getOptionalParameter(config, 'cloudfunctions_username')
+    password = getOptionalParameter(config, 'cloudfunctions_password')
     package = getRequiredParameter(config, 'cloudfunctions_package')
     namespaceUrl = getRequiredParameter(config, 'cloudfunctions_url')
     functionDir = getRequiredParameter(config, 'common_functions')
 
+    if apikey:
```

Comment:
Do we really want to permit to specify both username/password and apikey while silently ignoring username/password?
Anyway, you should update `test_badArgs` in `function_deploy` unit tests accordingly.

Reply:
It' also handled in `wawCommons.py`